src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -257,6 +259,33 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
         }
     }
 
+    async componentDidUpdate(prevProps: Props, prevState: ComponentState) {
+        this.handleQueryParameter(prevProps, prevState)
+    }
+
+    async handleQueryParameter(prevProps: Props, prevState: ComponentState): Promise<void> {
+        const searchParamsPrev = new URLSearchParams(prevProps.location.search)
+        const selectedDialogIdPrev = searchParamsPrev.get("id")
+
+        const searchParams = new URLSearchParams(this.props.location.search)
+        const selectedDialogId = searchParams.get("id")
+
+        if (selectedDialogId === selectedDialogIdPrev) {
+            return
+        }
+
+        if (selectedDialogId && 
+            (!this.state.isEditDialogModalOpen && !this.state.isTeachDialogModalOpen)) {
+            const trainDialog = this.props.trainDialogs.find(td => td.trainDialogId === selectedDialogId)
+            if (!trainDialog) {
+                // Invalid train dialog, go back to TD list
+                this.props.history.replace(`/home/${this.props.app.appId}/trainDialogs`, {app: this.props.app})
```

Comment:
Similar, should be able to use the `match.url` or something on the match object.

Reply:
I think it is `match.path` 
https://reacttraining.com/react-router/web/api/match